In [1]:
import warnings
warnings.filterwarnings("ignore")

In [2]:
import pandas as pd
data = pd.read_csv("LI_STAT_NOV.csv")
data.head()

Patient_id  Oxy_level  Pulse_rate  Blood_Sugar  Age Sex  Old_1  Old_2  \
0  COVPAT0001  97.109182  109.156925           83    5   M  30.08  23.14   
1  COVPAT0002  98.039265  109.481086          119    3   M  18.38  28.86   
2  COVPAT0003  97.904030  110.428601           80    8   F  30.27  24.32   
3  COVPAT0004  99.212383  110.048527           95    5   M  28.81  21.73   
4  COVPAT0005  99.235056  110.180046           82    6   M  25.06  32.65   

   Old_3  New_1  New_2  New_3  
0  33.58  30.51  28.75  22.47  
1  38.32  21.66  28.53  16.05  
2  42.16  21.99  31.26  20.15  
3  21.46  24.66  26.34  15.90  
4  27.54  25.84  20.59  21.65

In [3]:
data.shape

(5000, 12)

In [4]:
data.columns

Index(['Patient_id', 'Oxy_level', 'Pulse_rate', 'Blood_Sugar', 'Age', 'Sex',
       'Old_1', 'Old_2', 'Old_3', 'New_1', 'New_2', 'New_3'],
      dtype='object')

# 1.	Test for the Assumptions about the patients’ health record.

## Part 1.a : Oxy_level for the patients for should be 99

#### Step 1 : We are going to conduct the test of Oxy_level for a population, i.e. if the Population mean of Oxy_level is 99.

Considering the problem statement on mind the Hypothesis would be ;

*   Null Hypothesis : Population Oxy_level Mean = 99

*   Alternative Hypothesis : Population Oxy_level Mean ≠ 99

#### Step 2 : By the type of the problem or the hypothesis,we can say a 'One Sample t-test' is appropriate for the situation.

So before performing a 'One Sample t-test', we need to test the assumption of one sample t-test ,i.e. the population should follow normal distribution. 

#### Testing whether the Oxy_level follow normal distribution or not

(performing a shapiro test)

#### Hypothesis of shapiro test :

*   Null Hypothesis : Population follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution


In [5]:
from scipy.stats import shapiro
from scipy import stats
# normality test
stat, p = shapiro(data['Oxy_level'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample is from Gaussian Population (fail to reject H0)')
else:
	print('Samples is not from Gaussian Population (reject H0)')

Statistics=1.000, p=0.598
Sample is from Gaussian Population (fail to reject H0)


#### Conclussion : As the data follow Normal Distribution,now we are ready to perform a sample t-test.

In [6]:
print(stats.ttest_1samp(data['Oxy_level'], 99))
p = stats.ttest_1samp(data['Oxy_level'], 99)[1]
alpha = 0.05
if p < alpha :
  print('The Population Oxy_level Mean is not 99 (reject H0)')
else:
  print('The Population Oxy_level Mean is 99 (fail to reject H0)')

Ttest_1sampResult(statistic=-0.2010490370335417, pvalue=0.8406684101608832)
The Population Oxy_level Mean is 99 (fail to reject H0)


#### Conclusion :The data is from the population with Oxy_level mean as 99

## Part 1.b : Pulse_rate for the patients for should not be more than 100

#### Step 1 : We are going to conduct the test of Pulse_rate for a population, i.e. if the Population mean of Pulse_rate is 100.

Considering the problem statement on mind the Hypothesis would be ;

*   Null Hypothesis : Population Pulse_rate Mean ≤ 100

*   Alternative Hypothesis : Population Pulse_rate Mean > 100

#### Step 2 : By the type of the problem or the hypothesis,we can say a 'One Sample t-test' is appropriate for the situation.

So before performing a 'One Sample t-test', we need to test the assumption of one sample t-test ,i.e. the population should follow normal distribution. 

#### Testing whether the Pulse_rate follow normal distribution or not

(performing a shapiro test)

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution


In [7]:
from scipy.stats import shapiro
from scipy import stats
# normality test
stat, p = shapiro(data['Pulse_rate'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample is from Gaussian Population (fail to reject H0)')
else:
	print('Samples is not from Gaussian Population (reject H0)')

Statistics=1.000, p=0.892
Sample is from Gaussian Population (fail to reject H0)


#### Conclusion : As the data follow Normal Distribution,now we are ready to perform a sample t-test.

In [8]:
print(stats.ttest_1samp(data['Pulse_rate'], 100))
p = stats.ttest_1samp(data['Pulse_rate'], 100)[1]
alpha = 0.05
if p/2 < alpha :
  print('The Population Pulse_rate Mean is more than 100 (reject H0)')
else:
  print('The Population Pulse_rate Mean is not more than 100 (fail to reject H0)')

Ttest_1sampResult(statistic=706.2946125241093, pvalue=0.0)
The Population Pulse_rate Mean is more than 100 (reject H0)


#### Conclussion :The data is from the population with Pulse_rate mean is more 100

## Part 1.c : Blood_Sugar for the patients for should be 150

#### Step 1 : We are going to conduct the test of sugar level for a population, i.e. if the Population mean of Blood_Sugar level is 150.

Considering the problem statement on mind the Hypothesis would be ;

*   Null Hypothesis : Population Sugar Level Mean = 150

*   Alternative Hypothesis : Population Sugar Level Mean ≠ 150



#### Step 2 : By the type of the problem or the hypothesis,we can say a 'One Sample t-test' is appropriate for the situation.

So before performing a 'One Sample t-test', we need to test the assumption of one sample t-test ,i.e. the population should follow normal distribution. 

#### Testing whether the Suagar level follow normal distribution or not

(performing a shapiro test)

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution


In [9]:
from scipy.stats import shapiro
from scipy import stats
# normality test
stat, p = shapiro(data['Blood_Sugar'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample is from Gaussian Population (fail to reject H0)')
else:
	print('Samples is not from Gaussian Population (reject H0)')

Statistics=0.916, p=0.000
Samples is not from Gaussian Population (reject H0)


#### Conclusion : As the data does not follow Normal Distribution,we can not perform a sample t-test.In this case we might need to look into non-parametric test. Here we will be performing Wilcoxon sign rank test.

In [10]:
# hypothesized median
M_0 = 150

# calculate the difference between sample percentages and M_0
diff = data['Blood_Sugar'] - M_0

# perform wilcoxon signed rank test
# pass the differnces to the parameter, 'x'
# pass the two-tailed condition to the parameter, 'alternative'
test_stat, p_value = stats.wilcoxon(x = diff, alternative = 'two-sided')

# print the test statistic and corresponding p-value
print('Test statistic:', test_stat)
print('p-value:', p_value)
alpha = 0.05
if p_value < alpha :
  print('The Median Sugar Leve of Population is not 150 (reject H0)')
else:
  print('The Median Sugar Leve of Population is 150 (fail to reject H0)')

Test statistic: 6105689.5
p-value: 0.6485642832102604
The Median Sugar Leve of Population is 150 (fail to reject H0)


#### Conclusion :The data is from the population with median sugar level as 150

## Part 1.d : Average age for the patients for should be not less than  40

#### Step 1 : We are going to conduct the test of Age for a population, i.e. if the Population mean of Age is not less than 40.

Considering the problem statement on mind the Hypothesis would be ;

*   Null Hypothesis : Population Age Mean  ≥ 40

*   Alternative Hypothesis : Population Age Mean < 40


#### Step 2 : By the type of the problem or the hypothesis,we can say a 'One Sample t-test' is appropriate for the situation.

So before performing a 'One Sample t-test', we need to test the assumption of one sample t-test ,i.e. the population should follow normal distribution. 

#### Testing whether the Age follow normal distribution or not

(performing a shapiro test)

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution


In [11]:
from scipy.stats import shapiro
from scipy import stats
# normality test
stat, p = shapiro(data['Age'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample is from Gaussian Population (fail to reject H0)')
else:
	print('Samples is not from Gaussian Population (reject H0)')

Statistics=0.911, p=0.000
Samples is not from Gaussian Population (reject H0)


#### Conclussion : As the data does not follow Normal Distribution,we can not perform a sample t-test.In this case we might need to look into non-parametric test. Here we will be performing Wilcoxon sign rank test.

In [12]:
# hypothesized median
M_0 = 40

# calculate the difference between sample percentages and M_0
diff = data['Age'] - M_0

# perform wilcoxon signed rank test
# pass the differnces to the parameter, 'x'
# pass the two-tailed condition to the parameter, 'alternative'
test_stat, p_value = stats.wilcoxon(x = diff, alternative = 'two-sided')

# print the test statistic and corresponding p-value
print('Test statistic:', test_stat)
print('p-value:', p_value)
alpha = 0.05
if p_value/2 < alpha :
  print('The Median Age of Population is not less than 40 (reject H0)')
else:
  print('The Median Age of Population is less than 40 (fail to reject H0)')

Test statistic: 5191300.5
p-value: 1.3343427884792995e-20
The Median Age of Population is not less than 40 (reject H0)


#### Conclusion :The data is from the population with median Age is not less than 40

In [13]:
data.columns

Index(['Patient_id', 'Oxy_level', 'Pulse_rate', 'Blood_Sugar', 'Age', 'Sex',
       'Old_1', 'Old_2', 'Old_3', 'New_1', 'New_2', 'New_3'],
      dtype='object')

# Part 2 : Validate if there is any gender discrimintion while selecting the sample

In [14]:
# use 'value_counts()' to calculate the count for each category in the variable 'education' 
observed_value = data['Sex'].value_counts()

# observed values
observed_value

F    2544
M    2456
Name: Sex, dtype: int64

In [15]:
# use 'value_counts()' to calculate the count for each category in the variable 'education' 
exp_count = [ len(data) / 2 ] * 2 

# count of each category
exp_count

[2500.0, 2500.0]

In [16]:
# calculate the expected values for each category
# multiply the count by sample size and divide it by total number of observations in the population
expected_value = [ len(data) / 2 ] * 2

In [17]:
# create a list of observed values
print("Observed Values: ", list(observed_value))

# create a list of expected values
print("Expected Values: ", list(expected_value))

Observed Values:  [2544, 2456]
Expected Values:  [2500.0, 2500.0]


In [18]:
import numpy as np
# calculate the χ2-value for 90% of confidence level
# use 'stats.chi2.isf()' to find the χ2-value corresponding to the upper tail probability 'q'
# pass the value of 'alpha' to the parameter 'q', here alpha = 0.1
# pass the degrees of freedom to the parameter 'df' 
# use 'round()' to round-off the value to 4 digits
chi2_val = np.abs(round(stats.chi2.isf(q = 0.05, df = 1), 4))

print('Critical value for chi-square test:', chi2_val)

Critical value for chi-square test: 3.8415


i.e. if the chi-square value is greater than 3.8415 or p-value is less than 0.05 then we reject the null hypothesis.

In [19]:
from scipy.stats import chisquare

In [20]:
# use the 'chisquare()' to perform the goodness of fit test
# the function returns the test statistic value and corresponding p-value
# pass the observed values to the parameter, 'f_obs'
# pass the expected values to the parameter, 'f_exp'
test_stat, p_value = chisquare(f_obs = observed_value, f_exp = expected_value)

print('Test statistic:', test_stat)
print('p-value:', p_value)

Test statistic: 1.5488
p-value: 0.21331268082606386


#### Conclusion : The above output shows that the chi-square test statistic is less than 3.8415 and the p-value is greater than 0.05, thus, we accept the null hypothesis and conclude that there is no significant difference between the observed and expected values.That means there is no gender discrimination while selecting the sample

# Part 3 : There is no association between the gender and age group on whom the test was applied

In [21]:
data['Age_group'] = data['Age']

In [22]:
# given data
# total marks of students from group A
data['Age_group'][data['Age'] <= 12 ] = 'group A'

# total marks of students from group B
data['Age_group'][(data['Age'] > 12) & (data['Age'] <= 25)] = 'group B'

# total marks of students from group C
data['Age_group'][(data['Age'] > 25) & (data['Age'] <= 40)] = 'group C'

# total marks of students from group D
data['Age_group'][(data['Age'] > 40) & (data['Age'] <= 58)] = 'group D'

# total marks of students from group E
data['Age_group'][(data['Age'] > 58) & (data['Age'] <= 81)] = 'group E'

In [23]:
data['Age_group'].unique()

array(['group A', 'group D', 'group C', 'group B', 'group E'],
      dtype=object)

In [24]:
pd.crosstab(data['Sex'], data['Age_group'])

Age_group  group A  group B  group C  group D  group E
Sex                                                   
F              493      304      325      432      990
M              471      258      345      403      979

In [25]:
# observed values  
observed_value = pd.crosstab(data['Sex'], data['Age_group'])
#pd.pivot_table(data, values="value", index=["Sex"], columns="Age_group", aggfunc='count').values

In [26]:
observed_value

Age_group  group A  group B  group C  group D  group E
Sex                                                   
F              493      304      325      432      990
M              471      258      345      403      979

#### Performing a 2-way Chi-square test : to test the independence between the gender and age-group.

In [27]:
from scipy.stats import chi2_contingency
# use the 'chi2_contingency()' to check the independence of variables
# the function returns the test statistic value, corresponding p-value, degrees of freedom of the test and expected values
# pass the observed values to the parameter, 'observed'
# 'correction = False' will not apply the Yates' correction
test_stat, p, dof, expected_value = chi2_contingency(observed = observed_value)

# print the output
print("Test statistic:", test_stat)
print("p-value:", p)
print("Degrees of freedom:", dof)
print("Expected values:", expected_value)

Test statistic: 4.385410736868801
p-value: 0.3563518358703542
Degrees of freedom: 4
Expected values: [[ 490.4832  285.9456  340.896   424.848  1001.8272]
 [ 473.5168  276.0544  329.104   410.152   967.1728]]


#### Conclusion : There is no association between the gender and age group, which implies the drug test has been applied on a random basis without any discrination.

# 4.	Validate the claim that New Drug works better:


#### To validate this claim, We would like to have a stage wise  comparison between drugs.

## Test 4.a : Comparison test  between New Drug and Old Drug in Stage 1

*   Null Hypothesis : For stage 1 ;Mean performance of the Old Drug =  Mean performance of the New Drug
*   Alternative Hypothesis : For stage 1 ;Mean performance of the Old Drug ≠  Mean performance of the New Drug

#### From the problem statement it is very obvious that the test is a two sample independent t-test.

#### To consider the t-test, we first need to check 

*   whether the samples are from Normal population
*   Then samples are from homogeneous distribution



#### Step 1 : Normality test for performance of the Old Drug

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [28]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['Old_1'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample is from Gaussian Population (fail to reject H0)')
else:
	print('Samples is not from Gaussian Population (reject H0)')

Statistics=1.000, p=0.487
Sample is from Gaussian Population (fail to reject H0)


#### Conclussion : the data about performance of the Old Drug in Stage 1 follows Normal Distribution

#### Step 2 : Normality test for performance of the New Drug

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [29]:
# normality test
stat, p = shapiro(data['New_1'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample is from Gaussian Population (fail to reject H0)')
else:
	print('Sample is not from Gaussian Population (reject H0)')

Statistics=1.000, p=0.518
Sample is from Gaussian Population (fail to reject H0)


#### Conclusion : the data about performance of the New Drug in Stage 1 follows Normal Distribution

#### Step 3 : Homogeniety test for performance of the OLd Drug & performance of the New Drug

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populations are having same variance(homogeneous)
*   Alternative Hypothesis : Populations are not having same variance(not homogeneous)

In [30]:
from scipy import stats
#Null Hypothesis : var(performance of the old Drug) =var(performance of the new Drug)
#Alternative Hypothesis : var(performance of the old Drug) !=var(performance of the new Drug)
print(stats.levene(data['Old_1'],data['New_1']))
# interpret
p = stats.levene(data['Old_1'],data['New_1'])[1]
alpha = 0.05
if p > alpha:
	print('Populations are homogeneous (fail to reject H0)')
else:
	print('Populations are not homogeneous (reject H0)')

LeveneResult(statistic=0.5266897379212527, pvalue=0.46801888386946944)
Populations are homogeneous (fail to reject H0)


#### Conclusion : The performance of the old Drug and the new Drug in Stage 1 are from homogeneous population.Now as all the assumpion of two sample t-test are satisfied,we can procceed to perform two sample independent t-test.

#### Step 4 : Two sample t-test

Null Hypothesis : For stage 1 ;Mean performance of the new Drug =  Mean performance of the old Drug

Alternative Hypothesis : For stage 1 ;Mean performance of the new Drug ≠  Mean performance of the old Drug

In [32]:
from scipy import stats
## t-test
#Null Hypothesis is PopulationMean(performance of the old Drug) = PopulationMean(performance of the new Drug)
#Alternative Hypothesis is PopulationMean(performance of the old Drug) ≠ PopulationMean(performance of the new Drug)
t2, p2 = stats.ttest_ind(data['Old_1'],data['New_1'])
print("t-statistics is " + str(t2))
print("p-value is " + str(p2))
# interpret
alpha = 0.05
if p2 > alpha:
	print('Mean of the Populations are same (fail to reject H0)')
else:
	print('Mean of the Populations are not same (reject H0)')

t-statistics is -0.31003709333238755
p-value is 0.7565392107072749
Mean of the Populations are same (fail to reject H0)


#### Conclusion : The performance of the old Drug & the new Drug in Stage 1 are same.

#### We can not conclude anything about better performance among the old Drug & the new Drug from Stage 1 So we will procceed further into Stage 2 and would try to draw a comparative inference from there.

## Test 4.b : Comparison test  between the old Drug and the new Drug in Stage 2

*   Null Hypothesis : For stage 2 ;Mean performance of the old Drug =  Mean performance of the new Drug
*   Alternative Hypothesis : For stage 2 ;Mean performance of the old Drug ≠  Mean performance of the new Drug

#### From the problem statement it is very obvious that the test is a two sample independent t-test.

#### To consider the t-test, we first need to check 

*   whether the samples are from Normal population
*   Then samples are from homogeneous distribution


#### Step 1 : Normality test for performance of the old Drug

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [33]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['Old_2'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample looks Gaussian (fail to reject H0)')
else:
	print('Sample does not look Gaussian (reject H0)')

Statistics=0.999, p=0.056
Sample looks Gaussian (fail to reject H0)


#### Conclussion : the data about performance of the old Drug in Stage 2 follows Normal Distribution

#### Step 2 : Normality test for performance of the new Drug

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [34]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['New_2'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample looks Gaussian (fail to reject H0)')
else:
	print('Sample does not look Gaussian (reject H0)')

Statistics=1.000, p=0.763
Sample looks Gaussian (fail to reject H0)


#### Conclussion : the data about performance of the new Drug in Stage 2 follows Normal Distribution

#### Step 3 : Homogeniety test for performance of the old Drug & performance of the new Drug ;in Stage 2 

#### Hypothesis of levene test :

*   Null Hypothesis : Populations are having same variance(homogeneous)
*   Alternative Hypothesis : Populations are not having same variance(not homogeneous)

In [35]:
from scipy import stats
#Null Hypothesis : var(data1) =var(data2)
#Alternative Hypothesis : var(data1) ≠ var(data2)
print(stats.levene(data['Old_2'],data['New_2']))
# interpret
p = stats.levene(data['Old_2'],data['New_2'])[1]
alpha = 0.05
if p > alpha:
	print('Populations are homogeneous (fail to reject H0)')
else:
	print('Populations are not homogeneous (reject H0)')

LeveneResult(statistic=188.20579904706358, pvalue=1.8963351859772685e-42)
Populations are not homogeneous (reject H0)


#### Conclussion : The performance of the old Drug and the new Drug in Stage 2 are not from homogeneous population.Now as the assumption of two sample t-test is not getting satisfied,we need to think about non-parametric mean test.i.e. Mann-Whitney test.

In [36]:
print(stats.ttest_ind(data['Old_2'],data['New_2'],equal_var=False))
p = stats.ttest_ind(data['Old_2'],data['New_2'],equal_var=False)[1]
alpha = 0.05
if p > alpha :
    print('Both are having same population mean')
else:
    print('Samples are having different population mean')

Ttest_indResult(statistic=-11.682067948037085, pvalue=2.565661991049304e-31)
Samples are having different population mean


#### Conclussion : From the above Welch's t-test,we can see that the mean performance of the old Drug and the New Drug are not equal.Now we would like to see which Drug performs better.So instead of considering two-sided Mann-Whitney test ,we would perform a one-sided Mann-Whitney test.

#### Step 4 : Welch t-test, to check if he performance of the old Drug is better than the performance of the new Drug in stage 2

* Null Hypothesis : the performance of the old Drug is better than the performance of the new Drug in stage 2
* Alternative Hypothesis : the performance of the old Drug is inferior than the performance of the new Drug in stage 2

In [37]:
print(stats.ttest_ind(data['Old_2'],data['New_2'],equal_var=False))
p = stats.ttest_ind(data['Old_2'],data['New_2'],equal_var=False)[1]
alpha = 0.05
if p/2 > alpha :
    print('Mean of the old Drug is more than Mean of the new Drug  (fail to reject H0)')
else:
    print('Mean of the old Drug is less than Mean of the new Drug  (reject H0)')

Ttest_indResult(statistic=-11.682067948037085, pvalue=2.565661991049304e-31)
Mean of the old Drug is less than Mean of the new Drug  (reject H0)


#### Conclussion : the performance of the new Drug is better than the performance of the old Drug in stage 2

## Test 4.c : Comparison test  between the old Drug and the new Drug in Stage 3

*   Null Hypothesis : For stage 3 ;Mean performance of the old drug =  Mean performance of the new drug
*   Alternative Hypothesis : For stage 3 ;Mean performance of the old drug ≠  Mean performance of the new drug

#### From the problem statement it is very obvious that the test is a two sample independent t-test.

#### To consider the t-test, we first need to check 

*   whether the samples are from Normal population
*   Then samples are from homogeneous distribution

#### Step 1 : Normality test of the effect of the Old drug for stage 3

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [39]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['Old_3'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample looks Gaussian (fail to reject H0)')
else:
	print('Sample does not look Gaussian (reject H0)')

Statistics=0.999, p=0.047
Sample does not look Gaussian (reject H0)


#### Conclussion : the data about performance of the old Drug in Stage 3 does not follow Normal Distribution

#### Step 2 : Normality test of the effect of the new drug for stage 3

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [40]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['New_3'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
	print('Sample looks Gaussian (fail to reject H0)')
else:
	print('Sample does not look Gaussian (reject H0)')

Statistics=1.000, p=0.249
Sample looks Gaussian (fail to reject H0)


#### Conclussion : 

* The data about performance of the new Drug in Stage 3 follows Normal Distribution

* The performance of the old Drug in Stage3 does not follow Normal distribution . So the assumption of two sample t-test is not getting satisfied,we need to think about non-parametric mean test.i.e. Mann-Whitney test.

In [41]:
print(stats.mannwhitneyu(data['Old_3'],data['New_3']))
p = stats.mannwhitneyu(data['Old_3'],data['New_3'])[1]
alpha = 0.05
if p > alpha :
    print('Both are having same population median')
else:
    print('Samples are having different population median')

MannwhitneyuResult(statistic=563953.0, pvalue=0.0)
Samples are having different population median


#### Conclussion : From the above Mann-Whitney t-test,we can see that the mean performance of the old Drug and the New Drug are not equal.Now we would like to see which Drug performs better.So instead of considering two-sided Mann-Whitney test ,we would perform a one-sided Mann-Whitney test.

#### Step 3 : Mannwhitney Test : to check the performance of the old Drug is better than the performance of the new Drug in stage 3

#### Null Hypothesis : the performance of the old Drug is better than the performance of the new Drug in stage 3
#### Alternative Hypothesis : the performance of the old Drug is inferior than the performance of the new Drug in stage 3

In [42]:
print(stats.mannwhitneyu(data['Old_3'],data['New_3']))
p = stats.mannwhitneyu(data['Old_3'],data['New_3'])[1]
alpha = 0.05
if p/2 > alpha :
    print('Median of the old Drug is more than Median of the new Drug  (fail to reject H0)')
else:
    print('Median of the old Drug is less than Median of the new Drug  (reject H0)')

MannwhitneyuResult(statistic=563953.0, pvalue=0.0)
Median of the old Drug is less than Median of the new Drug  (reject H0)


#### Conclusion : the performance of the new Drug is better than the performance of the old Drug in stage 3

### Conclussion :
##### *   In stage 1 : The performance of the old Drug & the new Drug are same.
##### *   In stage 2 : The performance of the new Drug is better than the old Drug.
##### *   In stage 3 : The performance of the new Drug is better than the old Drug.

So,as a whole the new Drug performs better than the old Drug

# Test 5 : Validate the claim that there is gradual improvement in performance of the New Drug over the phases:

## Test 5.a : We will compare the result of the new Drug over stage 1 & Stage 2 ,first.

*   Null Hypothesis :The performance of Result-1 = The performance of Result-2
*   Alternative Hypothesis :The performance of Result-1 ≠ The performance of Result-2 

As in the data(effect of the new Drug) has been collected on the same patient(as in the same population) ,this is clearly the case of pair-wise t-test.

Before conducting the pair-wise t-test,we need to test whether the performance of Result-1 & Result-2 follows Normal distribution.

#### Step 1 : Normality test of the effect of the new drug for stage 1

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [44]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['New_1'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
    print('Sample looks Gaussian (fail to reject H0)')
else:
    print('Sample does not look Gaussian (reject H0)')

Statistics=1.000, p=0.518
Sample looks Gaussian (fail to reject H0)


#### Conclussion : the data about performance of the new Drug stored as Result 1 follows Normal Distribution


#### Step 2 : Normality test of the effect of the new drug for stage 2

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [45]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['New_2'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
    print('Sample looks Gaussian (fail to reject H0)')
else:
    print('Sample does not look Gaussian (reject H0)')

Statistics=1.000, p=0.763
Sample looks Gaussian (fail to reject H0)


#### Conclussion : the data about performance of the new Drug stored as Result 2 follows Normal Distribution.

#### Step 3 : Pair-wise t-test : now we can perform a pairwise t-teston Result-1 and Result-2.

In [46]:
print(stats.ttest_rel(data['New_1'], data['New_2']))
p = stats.ttest_rel(data['New_1'], data['New_2'])[1]
alpha = 0.05
if p < alpha :
    print('The treatment has significant effect (reject H0)')
else:
    print('The treatment has no significant effect (fail to reject H0)')

Ttest_relResult(statistic=1.241305421533698, pvalue=0.21455118521659772)
The treatment has no significant effect (fail to reject H0)


#### Conclussion : the performance of the new Drug in Result-1 & Result-2 have no significant difference

## Test 5.b : We will compare the result of the new Drug over stage 1 & Stage 3 ,first.

*   Null Hypothesis :The performance of Result-1 = The performance of Result-3
*   Alternative Hypothesis :The performance of Result-1 ≠ The performance of Result-3 

As in the data(effect of the new Drug) has been collected on the same patient(as in the same population) ,this is clearly the case of pair-wise t-test.

Before conducting the pair-wise t-test,we need to test whether the performance of Result-1 & Result-3 follows Normal distribution.

#### Step 1 : Normality test of the effect of the new drug for stage 1

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [47]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['New_1'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
    print('Sample looks Gaussian (fail to reject H0)')
else:
    print('Sample does not look Gaussian (reject H0)')

Statistics=1.000, p=0.518
Sample looks Gaussian (fail to reject H0)


#### Conclusion : the data about performance of the new Drug stored as Result 1 follows Normal Distribution

#### Step 2 : Normality test of the effect of the new drug for stage 3

#### Hypothesis of shapiro test :

*   Null Hypothesis : Populaton follow Normal Distribution
*   Alternative Hypothesis:Population does not follow Normal Distribution

In [48]:
from scipy.stats import shapiro
# normality test
stat, p = shapiro(data['New_3'])
print('Statistics=%.3f, p=%.3f' % (stat, p))
# interpret
alpha = 0.05
if p > alpha:
    print('Sample looks Gaussian (fail to reject H0)')
else:
    print('Sample does not look Gaussian (reject H0)')

Statistics=1.000, p=0.249
Sample looks Gaussian (fail to reject H0)


#### Conclusion : the data about performance of the new Drug stored as Result 3 follows Normal Distribution

#### Step 3 : Pair-wise t-test : now we can perform a pairwise t-teston Result-1 and Result-3.

#### We will compare the result of the new Drug over stage 1 & Stage 3 ,first.

*   Null Hypothesis :The performance of Result-1 = The performance of Result-3
*   Alternative Hypothesis :The performance of Result-1 ≠ The performance of Result-3 

As in the data(effect of the new Drug) has been collected on the same patient(as in the same population) ,this is clearly the case of pair-wise t-test.

In [49]:
print(stats.ttest_rel(data['New_1'], data['New_3']))
# Null Hypothesis: Result of Phase 1 is similar to Result of Phase 3
# Alternative Hypothesis: Result of Phase 1 is different than  Result of Phase 3
p = stats.ttest_rel(data['New_1'], data['New_3'])[1]
alpha = 0.05
if p < alpha :
    print('The treatment has significant effect (reject H0)')
else:
    print('The treatment has no significant effect (fail to reject H0)')

Ttest_relResult(statistic=88.05018179818472, pvalue=0.0)
The treatment has significant effect (reject H0)


#### Conclusion : the performance of the new Drug in Result-1 & Result-3 have significant difference.


## Test 5.c : Now we would like to see if the performance in Result 1 is better than the performance of Result 3 or not.

#### So we would like to consider the Hypothesis :

*   Null Hypothesis : Result-1 performs better than Result 3
*   Alternative Hypothesis : Result-1 does not perform better than Result 3

In [50]:
print(stats.ttest_rel(data['New_1'], data['New_3']))
# Null Hypothesis: Result of Phase 1 is better than  Result of Phase 3
# Alternative Hypothesis: Result of Phase 1 is worse than  Result of Phase 3
p = stats.ttest_rel(data['New_1'], data['New_3'])[1]
alpha = 0.05
if p/2 < alpha :
    print('Mean of effect in phase 1 is less than Mean of of effect in phase 3 (reject H0)')
else:
    print('Mean of effect in phase 1 is more than Mean of of effect in phase 3 (fail to reject H0)')

Ttest_relResult(statistic=88.05018179818472, pvalue=0.0)
Mean of effect in phase 1 is less than Mean of of effect in phase 3 (reject H0)


#### Conclusion :
Result of Phase 3 is better than  Result of Phase 1

Even though there was no significant difference in the performance of Result-1 & Result-2,there is a significant diffrence in the performance of Result-1 & Result-3.To be more specific there is an improvement in Result-3 as compared to Result-1

# Test 6 : Test the effect of the new drug over different age group 

In [51]:
# given data
# total marks of students from group A
gr_A = data[data['Age_group'] == 'group A']['New_3']

# total marks of students from group B
gr_B = data[data['Age_group'] == 'group B']['New_3']

# total marks of students from group C
gr_C = data[data['Age_group'] == 'group C']['New_3']

# total marks of students from group D
gr_D = data[data['Age_group'] == 'group D']['New_3']

# total marks of students from group E
gr_E = data[data['Age_group'] == 'group E']['New_3']

In [52]:
# perform Shapiro-Wilk test to test the normality
# shapiro() returns a tuple having the values of test statistics and the corresponding p-value
# pass the total marks of students 
stat, p_value = stats.shapiro(data['New_3'])

# print the p-value
print('p-value:', p_value)
# interpret
alpha = 0.05
if p_value > alpha:
    print('Sample looks Gaussian (fail to reject H0)')
else:
    print('Sample does not look Gaussian (reject H0)')

p-value: 0.24914191663265228
Sample looks Gaussian (fail to reject H0)


In [53]:
# perform Levene's test for the equality of variances 
# levene() returns a tuple having the values of test statistics and the corresponding p-value
# pass the total scores of students from each group
stat, p_value = stats.levene(gr_A, gr_B, gr_C, gr_D, gr_E)

# print the p-value 
print('P-Value:', p_value)
# interpret
alpha = 0.05
if p_value > alpha:
    print('the variance of effects of vaccine on different age group is not different (fail to reject H0)')
else:
    print('the variance of effects of vaccine on different age group is different. (reject H0)')

P-Value: 0.2916281058790177
the variance of effects of vaccine on different age group is not different (fail to reject H0)


In [54]:
# perform one-way ANOVA
# pass the given data
f_test, p_val = stats.f_oneway(gr_A, gr_B, gr_C, gr_D, gr_E)

# print the test statistic and corresponding p-value
print('Test statistic:', f_test)
print('p-value:', p_val)
# interpret
alpha = 0.05
if p_value > alpha:
    print('Average effect of vaccine on different age group is not different (fail to reject H0)')
else:
    print('Average effect of vaccine on different age group is different. (reject H0)')

Test statistic: 0.8102960966691605
p-value: 0.518369274029178
Average effect of vaccine on different age group is not different (fail to reject H0)


#### Conclusion :

Average effect of vaccine on different age group is not different 